sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java

```diff
@@ -320,4 +323,12 @@ public List<CosmosDiagnostics> getCancelledRequestDiagnosticsTracker() {
     public void setCancelledRequestDiagnosticsTracker(List<CosmosDiagnostics> cancelledRequestDiagnosticsTracker) {
         this.cancelledRequestDiagnosticsTracker = cancelledRequestDiagnosticsTracker;
     }
+
+    public void setPartitionKeyDefinition(PartitionKeyDefinition partitionKeyDefinition) {
```

Comment:
Why set the PKDefinition here in Public API?

Reply:
@FabianMeiswinkel - `CosmosQueryRequestOptionsImpl` is in the implementation package and this module is not opened up publicly. It is used by `CosmosQueryRequestOptions` but only as an encapsulation and not as inheritance so the public methods you see in `CosmosQueryRequestOptions` don't leak out in case of `setPartitionKeyDefinition` and `getPartitionKeyDefinition`.